### PR TITLE
feat(inference): OpenVINO backend for Perch v2 (ARM A76 CPU + Intel iGPU)

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Extends the native OpenVINO inference backend (added for BirdNET v2.4 in #3547) to the Google Perch v2 classifier, opt-in behind the `openvino` build tag with graceful fallback to ONNX Runtime. Default builds (no tag) are byte-for-byte unchanged.

Perch v2 runs as a secondary `ModelInstance` and applies softmax (unlike the primary BirdNET path, which applies sigmoid), so OpenVINO is wired into `NewPerch` (where Perch actually runs), not the primary `shouldTryOpenVINO` path. The two paths share a single device-plan helper.

### Backend changes
- **cgo binding**: per-classifier output port index (Perch's species logits are output index 3, behind `embedding`/`spatial_embedding`/`spectrogram`), CPU/GPU device selection, `INFERENCE_NUM_THREADS` omitted for the GPU plugin, a real output-dimension query from the compiled model, and Intel-GPU enumeration via `ov_core_get_available_devices`.
- **Output validation**: the OpenVINO classifier now reports `NumClasses` from the compiled model's selected output and validates it against the label count, so a model/label mismatch is caught at load and falls back to ORT instead of being masked (previously the count was derived from the label list, making the check tautological).
- **Device gating** (`openVINOPlanFor`): auto-selects the Intel iGPU when present, else the ARM A76 (ASIMDHP/f16) CPU; BirdNET v2.4 stays CPU-only; explicit override via a new `BirdNET.OpenVINODevice` config (`auto`/`cpu`/`gpu`, hot-reloadable). No frontend changes.
- Perch uses OpenVINO only for the `no_dft` model variant; the stock `perch_v2.onnx` cannot compile on OpenVINO (a dynamic-rank DFT op) and stays on ORT.

### Scope
Backend only. Shipping the `perch_v2_no_dft` model via the model gallery (arch-aware) and any frontend/UI for the device knob are follow-ups.

## Related Issues
Relates to #3239 (Intel iGPU offloading of Perch v2).
Builds on #3547.

## Test Plan
- [x] Builds clean with and without the `openvino` tag; `go vet -tags openvino` clean; default-build `golangci-lint` clean.
- [x] Unit tests pass with `-race` (gating, config validation/hot-reload, Perch no_dft detection, build-tag-split stub tests).
- [x] Lib-gated functional tests verify the index-3 logits output (14795 classes, not the 1536 embedding) and the NumClasses/label validation.
- [x] Output parity vs ONNX Runtime (top-5 identical) verified on amd64 CPU, amd64 Intel Iris Xe iGPU, and Raspberry Pi 5 (Cortex-A76, f16 CPU).

## Release Notes

* **New Features**
  * Added OpenVINO device preference (auto/cpu/gpu) for BirdNET inference and enabled hot-reload for device changes.
  * Expanded Perch v2 to use OpenVINO when compatible, with safe fallback to ONNX Runtime.
* **Improvements**
  * Improved settings handling by warning on unknown OpenVINO device values and falling back to auto.
* **Tests**
  * Added/expanded OpenVINO unit, functional, and backend smoke tests, including device enumeration and logits output validation, plus non-OpenVINO stub behavior checks.

### CODE REVIEW

File: internal/classifier/model_openvino.go [hunk 1]
```diff
@@ -11,24 +12,115 @@ import (
...
+
+// openVINOGPUAvailable reports whether an OpenVINO GPU device (Intel iGPU/dGPU via
+// the Intel GPU plugin) is present. It loads the OpenVINO core first (InitOpenVINO
+// is idempotent and fast on repeat); a load failure means no usable OV at all.
+func openVINOGPUAvailable(libraryPath string) bool {
+	if err := inference.InitOpenVINO(libraryPath); err != nil {
 		return false
 	}
+	return inference.OpenVINOHasDevice(inference.OVDeviceGPU)
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/classifier/model_openvino.go:101]: Good catch. Added a Warn log in openVINOGPUAvailable so a failed libopenvino_c load surfaces the real cause instead of the generic 'not eligible' fallback message. Fixed in e785b58d.

File: internal/inference/openvino/backend_openvino.go [hunk 2]
```diff
@@ -310,6 +325,88 @@ static void ovbind_infer_request_free(ov_infer_request_t* req) {
...
+        count *= d;
+    }
+    OVB.shape_free(&shape);
+
+    if (count <= 0) {
+        return OVBIND_ERR_BAD_OUTPUT;
+    }
+    *out_count = count;
+    return OK;
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/inference/openvino/backend_openvino.go:386]: Added an int64 overflow guard (count > INT64_MAX / d) before the multiply in ovbind_output_class_count, so a malformed or huge shape fails safe and falls back to ORT instead of wrapping. Fixed in e785b58d.

File: internal/classifier/perch_openvino_functional_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		OpenVINOPath:   os.Getenv("OV_TEST_LIB"),
+		Backend:        conf.BackendPrefOpenVINO,
+		OpenVINODevice: device,
+	}
+	t.Cleanup(func() { _ = inference.DestroyOpenVINO() })
+
+	c, ok := tryPerchOpenVINO(&cfg, labels)
+	require.True(t, ok, "Perch OpenVINO classifier must be created for the no_dft model")
+	require.NotNil(t, c)
+	t.Cleanup(func() { c.Close() })
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/classifier/perch_openvino_functional_test.go:57]: inference.Classifier.Close() returns no value (see internal/inference/backend.go), so there is no error to check here and the suggested `if err := c.Close()` would not compile. Leaving the cleanup as-is.

File: internal/classifier/orchestrator_perch_onnx.go [hunk 4]
```diff
@@ -39,9 +39,12 @@ func (o *Orchestrator) loadPerch(threads int) error {
 		LabelPath:       labelPath,
 		ONNXRuntimePath: o.Settings.BirdNET.ONNXRuntimePath,
 		Threads:         threads,
+		Backend:         o.Settings.BirdNET.Backend,
+		OpenVINOPath:    o.Settings.BirdNET.OpenVINOPath,
+		OpenVINODevice:  o.Settings.BirdNET.OpenVINODevice,
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/classifier/orchestrator_perch_onnx.go:44]: This is by design. ORT is the guaranteed fallback for Perch: tryPerchOpenVINO attempts OpenVINO only for the no_dft model and falls back to ORT on any failure, so it never makes Perch fail to load. The OpenVINO-capable image variants are built on top of the ONNX/ORT images, so ORT is always present; an OpenVINO-only deployment without ORT is not a supported configuration today, and keeping the ORT precheck guarantees the fallback path exists.

**Comment by tphakala** [comment]
@coderabbitai review
